docs/docs/alephium/collections.md

```diff
@@ -0,0 +1,596 @@
+| ----- | ------- | --------------------- |
+| chunk   | I256 | Index of a tick.|
+| poolKey | poolKey     | The pool key of a specified pool.|
+
+#### Output parameters
+
+| Type | Description                 |
+| ---- | --------------------------- |
+| U256 | The chunk's bits. |
+
+### Flip bit utility
+
+```rust
+fn flipBitAtPosition(value: U256, position: U256) -> U256;
+```
+
+Flips bit at position.
+
+| Name  | Type    | Description           |
+| ----- | ------- | --------------------- |
+| value   | U256 | The value in which we flip a bit.|
+| position | U256     | The position of a bit to flip.|
+
+#### Output parameters
+
+| Type | Description                 |
+| ---- | --------------------------- |
+| U256 | `value` variable with the specified bit flipped. |
+
+### Contains initialized tick
+
+```rust
+pub fn getBit(tick: I256, poolKey: PoolKey) -> Bool;
+```
+
+Retrieves the state of the exact bit representing the tick.
+
+
+#### Input parameters
+
+| Name  | Type    | Description           |
+| ----- | ------- | --------------------- |
+| tick   | I256 | Index of a tick.|
+| poolKey | poolKey     | The pool key of a specified pool.|
+
+#### Output parameters
+
+| Type | Description                 |
+| ---- | --------------------------- |
+| Bool | State of the tick, `true` if initialized. |
+
+## Reserves
+
+```rust
+Abstract Contract Reserves() {...}
+
+Contract Invariant(...) extends Reserves(), ...{
+    ...
+        // Key: TokenId, Value: ReserveId
+    mapping[ByteVec, ByteVec] reserves
+    ...
+}
+```
+
+The `Reserves` Abstract Contract is designed to overcome the challenge of being able to store a limited number of assets (tokens) in a single contract. It utilizes a mapping data structure where each token's `Reserve`'s location is identified by the id of a given asset. The need to introduce a collection like this is due to the limited number of assets a single UTXO can store, currently 8. Each Contract has a single UTXO. We went with this design because if both tokens for a trading pair are already in a reserve users pay only for storage of trading pair specific information.
+
+| Type                   | Key                               | Value                                 |
+| ---------------------- | --------------------------------- | ------------------------------------- |
+| mapping[ByteVec, ByteVec] | The ContractId of a given token. | The ContractId of the reserve containing the given token. |
+
+### Add reserve
+
+```rust
+@using(updateFields = true, preapprovedAssets = true)
+fn initReserve(caller: Address, reservePath: ByteVec, assetsToStore: U256) -> ByteVec;
+```
+
+Adds a new `Reserve` and instantly registers `assetsToStore` assets.
+
+
+#### Input parameters
+
+| Name  | Type    | Description           |
+| ----- | ------- | --------------------- |
+| caller   | Address | Address of the user who wants to know where the Token is stored. They pay the eventual fee.|
```

Comment:
They won't know where it is stored using this function. `Caller` is requried to pay the init fee